### ISSUE TITLE
Fatal error trying to parse sample XML doc

### ISSUE THREAD

**Issue by danjac** [manukall/feeder_ex#15]
```
iex(1)>  s = """<?xml version="1.0"?>
<rss version="2.0">
   <channel>
      <title>Liftoff News</title>
      <link>http://liftoff.msfc.nasa.gov/</link>
      <description>Liftoff to Space Exploration.</description>
      <language>en-us</language>
      <pubDate>Tue, 10 Jun 2003 04:00:00 GMT</pubDate>
      <lastBuildDate>Tue, 10 Jun 2003 09:41:01 GMT</lastBuildDate>
      <docs>http://blogs.law.harvard.edu/tech/rss</docs>
      <generator>Weblog Editor 2.0</generator>
      <managingEditor>editor@example.com</managingEditor>
.....(rest of doc)
"""

iex(2)> FeederEx.parse s

{:fatal_error,
 {:EXIT,
  {:function_clause,
   [{FeederEx.Parser, :event,
     [entry: {:entry, :undefined, :undefined, :undefined, :undefined,
       "http://liftoff.msfc.nasa.gov/2003/06/03.html#item573", :undefined,
       "http://liftoff.msfc.nasa.gov/news/2003/news-starcity.asp", :undefined,
       "How do Americans get ready to work with Russians aboard the International Space Station? They take a crash course in culture, language and protocol at Russia's <a href=\"http://howe.iki.rssi.ru/GCTC/gctc_e.htm\">Star City</a>.",
       "Star City", "Tue, 03 Jun 2003 09:39:21 GMT"}, nil: []],
     [file: 'lib/feeder_ex/parser.ex', line: 3]},
    {:feeder, :end_element, 2, [file: 'src/feeder.erl', line: 178]},
    {:xmerl_sax_parser_utf8, :event_callback, 2,
     [file: 'xmerl_sax_parser_utf8.erl', line: 3343]},
    {:xmerl_sax_parser_utf8, :parse_etag_1, 3,
     [file: 'xmerl_sax_parser_utf8.erl', line: 1116]},
    {:xmerl_sax_parser_utf8, :parse_document, 2,
     [file: 'xmerl_sax_parser_utf8.erl', line: 236]},
    {:xmerl_sax_parser_utf8, :parse, 2,
     [file: 'xmerl_sax_parser_utf8.erl', line: 159]},
    {:erl_eval, :do_apply, 6, [file: 'erl_eval.erl', line: 670]},
    {:elixir, :erl_eval, 3, [file: 'src/elixir.erl', line: 224]}]}}}

```
mix deps:

  defp deps do
    [
      {:feeder_ex, git: "https://github.com/manukall/feeder_ex.git"}
    ]
  end

Also tried {:feeder_ex, "~> 1.0.1"}

**Comment by danjac** [comment]
Further info: this is using Elixir 1.4.

**Comment by denic** [comment]
Works for me with the latest version in master.  Thanks guys!

Version 1.0 did not work with elixir 1.4.1 but did with 1.2.

**Comment by jomei** [comment]
Doesn't work for me. Tried to run the code from example
```elixir
{:ok, feed, _} = FeederEx.parse(body)
** (throw) {:EXIT, {:function_clause, [{FeederEx.Parser, :event, [entry: {:entry, :undefined, :undefined, :undefined, :undefined, "http://liftoff.msfc.nasa.gov/2003/06/03.html#item573", :undefined, "http://liftoff.msfc.nasa.gov/news/2003/news-starcity.asp", :undefined, "How do Americans get ready to work with Russians aboard the International Space Station? They take a crash course in culture, language and protocol at Russia's <a href=\"http://howe.iki.rssi.ru/GCTC/gctc_e.htm\">Star City</a>.", "Star City", "Tue, 03 Jun 2003 09:39:21 GMT"}, nil: []], [file: 'lib/feeder_ex/parser.ex', line: 3]}, {:feeder, :end_element, 2, [file: 'src/feeder.erl', line: 178]}, {:xmerl_sax_parser_utf8, :event_callback, 2, [file: 'xmerl_sax_parser_utf8.erl', line: 3309]}, {:xmerl_sax_parser_utf8, :parse_etag_1, 3, [file: 'xmerl_sax_parser_utf8.erl', line: 1075]}, {:xmerl_sax_parser_utf8, :parse_document, 2, [file: 'xmerl_sax_parser_utf8.erl', line: 183]}, {:xmerl_sax_parser_utf8, :parse, 2, [file: 'xmerl_sax_parser_utf8.erl', line: 115]}, {:erl_eval, :do_apply, 6, [file: 'erl_eval.erl', line: 670]}, {:erl_eval, :expr, 5, [file: 'erl_eval.erl', line: 438]}, {:elixir, :erl_eval, 3, [file: 'src/elixir.erl', line: 224]}, {:elixir, :eval_forms, 4, [file: 'src/elixir.erl', line: 212]}, {IEx.Evaluator, :handle_eval, 6, [file: 'lib/iex/evaluator.ex', line: 182]}, {IEx.Evaluator, :do_eval, 4, [file: 'lib/iex/evaluator.ex', line: 175]}, {IEx.Evaluator, :eval, 4, [file: 'lib/iex/evaluator.ex', line: 155]}, {IEx.Evaluator, :loop, 3, [file: 'lib/iex/evaluator.ex', line: 61]}, {IEx.Evaluator, :init, 4, [file: 'lib/iex/evaluator.ex', line: 21]}, {:proc_lib, :init_p_do_apply, 3, [file: 'proc_lib.erl', line: 247]}]}}
    xmerl_sax_parser_utf8.erl:131: :xmerl_sax_parser_utf8.parse/2
```

mac OS x/ elixir 1.4.1 / Erlang 19
Package ver: 1.0.1

**Comment by maxguzenski** [comment]
feeder_ex depends on feeder (a erlang package), and feeder_ex is not working with feeder 2.2.1, only with 2.1.0. Maybe @michaelnisi can help.

**this works**
```elixir
defp deps do
     [{:feeder, "2.1.0"},
     {:feeder_ex, "1.0.1"}]
end
```

**this not**
```elixir
defp deps do
     [{:feeder, "2.2.1"},
     {:feeder_ex, "1.0.1"}]
end
```

**Comment by michaelnisi** [comment]
I can parse said [feed](https://raw.githubusercontent.com/manukall/feeder_ex/master/test/sample-rss-2.xml) with the feeder [example](https://github.com/michaelnisi/feeder/tree/master/example):

```
% erl -pa deps/*/ebin ebin                                                                                    !10035
Erlang/OTP 19 [erts-8.1] [source] [64-bit] [smp:8:8] [async-threads:10] [kernel-poll:false]

Eshell V8.1  (abort with ^G)
1> ssl:start().
ok
2> example:start().
ok
3> example:print_titles("https://raw.githubusercontent.com/manukall/feeder_ex/master/test/sample-rss-2.xml").
Star City
undefined
The Engine That Does More
Astronauts' Dirty Laundry
ok
```

2.2.x added categories:

```
± % git diff v2.1.0 v2.2.0 | grep categories                                                                                                                                                                                           !10054
+- `Key = author | categories | duration | enclosure | id | image | link | subtitle | summary | title | updated`
+  append(E, #entry.categories, State#state.chars);
+get(categories, E) ->
+  E#entry.categories;
+  categories :: undefined | [binary()],
```

@adsteel might have an idea.

**Comment by adsteel** [comment]
@maxguzenski @danjac @manukall I noticed that `feeder_ex` hasn't bumped versions recently. Do those tests fail if you point `feeder_ex` at commit sha `5cab58f8110935e7412e0f7e7f236e85b81471a4` ? This error seems like it could come from `feeder_ex` expecting a different order of tuple items than is returned by `feeder` 2.2.1, which I would expect if the new `categories` changes aren't in the `feeder_ex` code being used.

**Comment by adsteel** [comment]
@danjac when I pull `feeder_ex` and run your example, everything works fine with both `feeder` 2.2.0 and 2.2.1.

```elixir
  s = """
<?xml version="1.0"?>
<rss version="2.0">
   <channel>
      <title>Liftoff News</title>
    ...(rest of example)
  """

{:ok, feed, _} = FeederEx.parse(s)

IO.inspect feed

=> %FeederEx.Feed{author: nil,
 entries: [%FeederEx.Entry{author: nil, categories: ["Current Events", "News"],
   duration: nil, enclosure: nil,
   id: "http://liftoff.msfc.nasa.gov/2003/06/03.html#item573", image: nil,
   link: "http://liftoff.msfc.nasa.gov/news/2003/news-starcity.asp",
   subtitle: nil,
  ...
```

**Comment by adsteel** [comment]
I can reproduce the error when I go back to `feeder_ex` version 1.0.1 and force `feeder` to 2.2.1. @manukall Looks like we just need a new version bump/publish. https://github.com/manukall/feeder_ex/pull/16

**Comment by manukall** [comment]
i published version 1.1 which should solve this. thanks @adsteel!
